### PR TITLE
feat: custom `minify` function

### PR DESCRIPTION
**Pull request by evilebottnawi**
Allow to use `uglify-js` and `terser`.

Next steps:
1. Update `webpack-defaults`.
2. Refactor.
3. Switch on `terser.`, update tests and docs.
4. Major release (drop support `nodejs@4`).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ module.exports = {
 |**`cache`**|`{Boolean\|String}`|`false`|Enable file caching|
 |**`parallel`**|`{Boolean\|Number}`|`false`|Use multi-process parallel running to improve the build speed|
 |**`sourceMap`**|`{Boolean}`|`false`|Use source maps to map error message locations to modules (This slows down the compilation) ⚠️ **`cheap-source-map` options don't work with this plugin**|
+|**`minify`**|`{Function}`|`(file, uglifyOptions) => uglify.minify(file, uglifyOptions)`|Allows you to override default minify function|
```
Comment (ooflorent) [inline README.md]: The second param should be named differently

Comment (shellscape) [reply README.md]: seconded. (this repo could really benefit from a webpack-defaults update)

Comment (evilebottnawi) [reply README.md]: `uglifyOptions` -> `options`?

Comment (ooflorent) [reply README.md]: `options` seems good

Comment (evilebottnawi) [reply README.md]: @ooflorent maybe better minifyOptions, because it will be misleading between options for plugin and options for minification

Comment (shellscape) [reply README.md]: yeah I could see that being a point of possible confusion

**Comment by evilebottnawi** [comment]
Close in favor https://github.com/webpack-contrib/uglifyjs-webpack-plugin/pull/325

**Pull request closed without merging**